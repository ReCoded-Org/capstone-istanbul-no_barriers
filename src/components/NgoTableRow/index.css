td.ngoDetails {
  padding-right: 4rem;
  padding-bottom: 3rem;
}

.btnMoreDetails,
.btnMoreDetails:hover,
.btnMoreDetails:focus {
  background-color: var(--orange);
  border: none;
}

/* filter all button active state style */
.btnMoreDetails.btn-primary:not(:disabled):not(.disabled):active {
  background-color: var(--dark-orange);
}