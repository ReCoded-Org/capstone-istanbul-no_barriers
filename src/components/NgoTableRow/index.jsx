import React, { useState } from "react";
import { Button } from "react-bootstrap";
import "./index.css";
import NgoDetails from "../NgoDetails";
import NgoDetailsDirection from "../NgoDetailsDirection";
import { useTranslation } from "react-i18next";

const NgoTableRow = ({
  trClassname,
  id,
  image,
  name,
  service,
  city,
  rating,
  details,
}) => {
  const [openDetails, setOpenDetails] = useState(false);
  const { t } = useTranslation();

  return (
    <>
      <tr className={trClassname} key={id}>
        <td className="d-table-cell align-middle">
          <img src={image} alt="NGO Logo" className="ngoLogo" align="right" />
        </td>
        <td className="d-table-cell align-middle">{name}</td>
        <td className="d-table-cell align-middle">{service}</td>
        <td className="d-table-cell align-middle">{city}</td>
        <td className="d-table-cell align-middle">{rating}</td>
        <td className="d-table-cell align-middle btnCol">
          <Button
            className="btnMoreDetails shadow-none"
            onClick={() => setOpenDetails(!openDetails)}
            aria-controls={`${id}`}
          >
            {t("resources.ngoList.buttonOpenDetails")}
          </Button>
        </td>
      </tr>
      {openDetails ? (
        <tr className="table-info">
          <td>&nbsp;</td>
          <td className="ngoDetails" id={`${id}`} colSpan="3">
            <NgoDetails details={details} />
          </td>
          <td colSpan="2">
            <NgoDetailsDirection address={details.address} />
          </td>
        </tr>
      ) : null}
    </>
  );
};

export default NgoTableRow;
