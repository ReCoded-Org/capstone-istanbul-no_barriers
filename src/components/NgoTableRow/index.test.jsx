import React from "react";
import renderer from "react-test-renderer";
import NgoTableRow from "./index";

it("NgoTableRow renders correctly", () => {
  const allNgos = [
    {
      city: "Ankara",
      details: {
        address: "Sincan/Ankara",
        email: "abc@admin.com",
        facilityName: "ABC schools",
        phoneNumber: "+90 000 000 00",
        responsiblePerson: "Aişe Şen",
      },
      serviceDescription: {
        arabic: "arabic",
        english:
          "We are helping. Suspendisse pulvinar pretium scelerisque. Suspendisse potenti. Sed ullamcorper enim sed lacinia sagittis. Nulla tellus risus, maximus et vestibulum id, tempor eget ipsum. Praesent convallis dictum turpis eu ullamcorper. Curabitur in ultrices sapien. Etiam lacinia ante quis luctus luctus. Nullam efficitur turpis orci, at bibendum tortor efficitur sed. Nam accumsan dolor luctus blandit lobortis.",
        turkish: "yardım",
      },
      website: "www.abc.net",
      image:
        "https://firebasestorage.googleapis.com/v0/b/no-barriers.appspot.com/o/Zootopia-Characters-Revealed-5.jpg?alt=media&token=0a108e5a-decf-4278-9284-094db45a160a",
      name: "ABC",
      rating: 2,
      service: "Education",
    },
  ];

  const tree = renderer
    .create(
      allNgos.map((ngo, index) => {
        if (index % 2 === 0) {
          return (
            <NgoTableRow
              trClassname="table-info"
              key={ngo.id}
              image={ngo.image}
              name={ngo.name}
              service={ngo.service}
              city={ngo.city}
              rating={ngo.rating}
              details={ngo.details}
            />
          );
        }
        return (
          <NgoTableRow
            trClassname="table-light"
            key={ngo.id}
            image={ngo.image}
            name={ngo.name}
            service={ngo.service}
            city={ngo.city}
            rating={ngo.rating}
            details={ngo.details}
          />
        );
      })
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
