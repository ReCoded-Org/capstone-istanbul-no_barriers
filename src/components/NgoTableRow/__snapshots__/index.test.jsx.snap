// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgoTableRow renders correctly 1`] = `
<tr
  className="table-info"
>
  <td
    className="d-table-cell align-middle"
  >
    <img
      align="right"
      alt="NGO Logo"
      className="ngoLogo fluid"
      src="https://firebasestorage.googleapis.com/v0/b/no-barriers.appspot.com/o/Zootopia-Characters-Revealed-5.jpg?alt=media&token=0a108e5a-decf-4278-9284-094db45a160a"
    />
  </td>
  <td
    className="d-table-cell align-middle"
  >
    ABC
  </td>
  <td
    className="d-table-cell align-middle"
  >
    Education
  </td>
  <td
    className="d-table-cell align-middle"
  >
    Ankara
  </td>
  <td
    className="d-table-cell align-middle"
  >
    2
  </td>
  <td
    className="d-table-cell align-middle btnCol"
  >
    <button
      aria-controls="undefined"
      className="btnMoreDetails shadow-none btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      resources.ngoList.buttonOpenDetails
    </button>
  </td>
</tr>
`;
